vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,902 @@
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+
+vgpr_y_imwg <- event(
+  description = "Define sCR, CR or VGPR as response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("sCR", "CR", "VGPR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(vgpr_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "VGPRRSP",
+      PARAM = "IMWG VGPR Rate by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSORRES, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD %in% c("CRRSP", "VGPRRSP") & AVALC == "Y"
+)
+```
+
+## Derive Best Overall Response Parameter {#bor}
+
+The function `admiral::derive_extreme_event()` can be used to derive the best
+overall response (without confirmation required) parameter.
+
+Please note that the order of the events specified for `events` is important.
+For example, a subject with `PR`, `PR`, `CR` qualifies for both `bor_cr` and
+`bor_pr`. As `bor_cr` is listed before `bor_pr`, `CR` is selected as best overall
+response for this subject.
+
+Some events such as `bor_cr`, `bor_pr` have been defined in {admiralonco} for
+RECIST 1.1. Missing events specific to IMWG criteria are defined below.  
+**Note**: *For `SD`, it is not required as for RECIST1.1 that the response occurs after a protocol-defined number of days.*
+
+```{r}
+bor_scr <- event(
+  description = "Define stringent complete response (sCR) for best overall response
+  (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "sCR",
+  set_values_to = exprs(AVALC = "sCR")
+)
+
+bor_vgpr <- event(
+  description = "Define very good partial response (VGPR) for best overall response
+  (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "VGPR",
+  set_values_to = exprs(AVALC = "VGPR")
+)
+
+bor_mr <- event(
+  description = "Define minimal response (MR) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "MR",
+  set_values_to = exprs(AVALC = "MR")
+)
+
+bor_sd_imwg <- event(
+  description = "Define stable disease (SD) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "SD",
+  set_values_to = exprs(AVALC = "SD")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr, ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(
+      bor_scr, bor_cr, bor_vgpr, bor_pr, bor_mr, bor_sd_imwg, bor_pd,
```

Comment:
Just want to make sure we will separate parameters (PARAM,PARAMCD) for VGPRR or better, CCR for CR or better etc correct?

Reply:
I derive VGPRRSP - "IMWG VGPR Response by Investigator" instead of VPGRR - "VGPR Response Rate" as written in work flow. It is consistent with previous vignettes imo.
Same for CRR -> CRRSP, ORR -> RSP.